slave/components/quality_controller/adapter_trimming.go

```diff
@@ -2,32 +2,126 @@ package quality_controller
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
 	"github.com/docker/docker/client"
 
 	"github.com/genomagic/config_parser"
+	"github.com/genomagic/constants"
 )
 
 // adapterTrimming is the struct representation of the adapter trimming process
 type adapterTrimming struct {
-	// context of the process
-	ctx context.Context
 	// dockerCLI is used for launching a Docker container that perform adapter trimming
 	dockerCLI *client.Client
 	// config is the GenoMagic global configuration
 	config *config_parser.Config
+	// the context of the process
+	ctx context.Context
 }
 
-// NewAdapterTrimming constructs and returns a new instance of adapterTrimming, which implements the Controller interface
 func NewAdapterTrimming(ctx context.Context, dockerCli *client.Client, config *config_parser.Config) Controller {
 	return &adapterTrimming{
-		ctx:       ctx,
 		dockerCLI: dockerCli,
 		config:    config,
+		ctx:       ctx,
 	}
 }
 
-// Process performs the adapter trimming process by launching a Docker container that uses Trimmomatic
+// TODO: Replace this function into a constants file where this can be used by other quality control processes
+// getImageID attempts to find the Docker image by given term
+func getImageID(client *client.Client, ctx context.Context, term string) (string, error) {
```

Comment:
For now, what do you think about placing this inside a `utils.go`, placed inside the `quality_control` package? It would make this file shorter and exclusively focused on adapter trimming logic, whereas the `utils.go` can contain any package private functions shared by the quality control processes.